src/airtable.js

```diff
@@ -113,5 +113,30 @@ exports.findVolunteerById = async id => {
   return base("Volunteers").find(id);
 };
 
+exports.findPaymentByCode = async (code) => {
+  try {
+    const record = await base("Requests")
+      .select({
+        // hate these hard-codes, consider getting from a static reference elsewhere or something
+        filterByFormula: `And({Code} = '${code}', {Status}!="Completed", {Status}!="FailedNoAnswer", {Status}!="FailedDonorBackedOut")`
+      })
+      .firstPage();
+    return record
+      ? [record[0], null]
+      : [null, "Valid payment with that code not found"];
+  } catch (e) {
+    console.error(`Error while fetching request by code: ${code}`); // TODO cargo culted from above, what is this rescuing? hopefully not just a network error...
```

Comment:
would love comments on this -- wasn't sure what this rescue was for, if it's something specific we should probably handle that specific thing? else at least log info about the error?